### PR TITLE
Restructuring the installation of libsodium

### PR DESCRIPTION
**Pull request by kkwoker**
Restructuring the installation of libsodium

### CODE REVIEW

File: tasks/install.yml [hunk 1]
```diff
@@ -93,10 +94,29 @@
...
     - 'cd /tmp/libsodium/ && make install'
-    - 'php5enmod libsodium'
+  when: php_sodium
+  tags:
+    - sodium
+
+- name: Sodium | Install PHP extension
+  sudo: yes
+  ignore_errors: yes
+  shell: pecl install channel://pecl.php.net/libsodium-1.0.2
```
Comment (noqcks) [inline tasks/install.yml]: why is this needed?

File: tasks/install.yml [hunk 2]
```diff
@@ -84,6 +84,7 @@
   git:
     repo: https://github.com/jedisct1/libsodium
     dest: /tmp/libsodium/
+    version: 1.0.8
```
Comment (noqcks) [inline tasks/install.yml]: why pulling a specific version of libsodium? couldn't we just pull the latest release from here? https://download.libsodium.org/libsodium/releases/

**Comment by noqcks** [comment]
@tinyroy can you answer some of the questions here, since you pushed all these changes :smiley_cat:

**Comment by tinyroy** [comment]
@kkwoker, keep the version hard coded. We don't control the dependency between libsodium and libsodium-php.

**Comment by noqcks** [comment]
@tinyroy is libsodium-php tied to specific versions of libsodium?

File: tasks/install.yml (hunk 1 above)
Comment (kkwoker) [inline tasks/install.yml]: what should it be?

Comment (noqcks) [inline tasks/install.yml]: my bad. disregard. thought this was libsodium version, not libsodium-php.

File: tasks/install.yml [hunk 3]
```diff
@@ -110,13 +130,9 @@
...
 
-- name: Sodium | Install PHP extension
+- name: "Sodium | php5enmod libsodium"
   sudo: yes
-  ignore_errors: yes
-  shell: pecl install channel://pecl.php.net/libsodium-1.0.2
-  notify:
-    - Reload Service | nginx
-    - service | php5-fpm | reloaded
+  shell: "php5enmod libsodium"
```
Comment (noqcks) [inline tasks/install.yml]: no when statement for `when: php_sodium`

Comment (kkwoker) [inline tasks/install.yml]: it's on line 136

**Comment by noqcks** [comment]
can you fix these merge conflicts @kkwoker ill pull it in

File: tasks/install.yml (hunk 2 above)
Comment (aaronpederson) [inline tasks/install.yml]: @kkwoker can you address this when possible?

Comment (kkwoker) [inline tasks/install.yml]: We dont know control the dependency between libsodium and libsodium-php, the question was answered in the conversation section
